File: Web/ClinicAI/Models/ClinicalNote.cs
using System;

namespace ClinicAI.Models
{
    public class ClinicalNote
    {
        public int Id { get; set; }
        public int CaseId { get; set; }
        public string Subjective { get; set; } = string.Empty;
        public string Objective { get; set; } = string.Empty;
        public string Assessment { get; set; } = string.Empty;
        public string Plan { get; set; } = string.Empty;
        public string? AdditionalInformation  { get; set; }
        public bool GeneratedByAI { get; set; }
        public bool ApprovedByDoctor { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation property
        public Case? Case { get; set; }
    }
}

File: Mobile_App/lib/features/clinical_notes/domain/entities/clinical_note.dart
import 'package:equatable/equatable.dart';

class ClinicalNote extends Equatable {
  final String id;
  final String subjective;
  final String objective;
  final String assessment;
  final String plan;
  final String additionalInformation;
  final DateTime date;
  final String patientName;

  const ClinicalNote({
    required this.id,
    required this.subjective,
    required this.objective,
    required this.assessment,
    required this.plan,
    required this.additionalInformation,
    required this.date,
    required this.patientName,
  });

  @override
  List<Object?> get props => [id, subjective, objective, assessment, plan, additionalInformation, date, patientName];
}
